Merge code and history of GPOS into ORCA

d opened the pull request:
This PR merges both code and history of the GPOS library into ORCA repo.

We've changed

0. The build system of ORCA to
  0. build GPOS as part of ORCA
  0. run GPOS tests as part of ORCA tests
0. The ORCA pipeline to eliminate parts pertaining to GPOS
0. README
0. Bumped ORCA version from 1.697 to 2.0

To reviewers, the Github PR UI will have too much noise, it'll be far easier to `git fetch` our branch to see relevant commits in your local Git environment.

This pull request supersedes #130

hsyuan commented:
+1

dhanashreek89 commented:
Looks like the `version.h.in` still exists underneath the `libgpos`, and should be removed also.

vraghavan78 commented on cmake/FindAtomics.cmake:
```diff
@@ -0,0 +1,81 @@
+# Copyright (c) 2015, Pivotal Software, Inc.
```
Is this a new file?

hsyuan replied:
no

d replied:
No venky, this is gotten from GPOS. We use this to detect support for atomic intrinsics. Currently GPOS uses a (somewhat portable, but less hip) implementation of atomics that's somehow tied to GCC (but it's also somewhat widely supported). This is necessary for our implementation of `gpos::UllExchangeAdd` and friends.

vraghavan78 commented on libgpos/targzs/ivysettings.xml:
```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
do we need this?

d replied:
will remove this entire folder

vraghavan78 commented on libgpos/targzs/ivy-devel.xml:
```diff
@@ -0,0 +1,32 @@
+<ivy-module version="2.0">
```
do we need this?

hsyuan replied:
the whole folder can be removed.

vraghavan78 commented on libgpos/targzs/ivy.xml:
```diff
@@ -0,0 +1,32 @@
+<ivy-module version="2.0">
```
do we need this>

d commented:
@hsyuan and I added two commits to address the above comments, specifically we

0. Removed the GPOS version, and with it, the generated version header
0. Removed unused artifact xml files

@vraghavan78 @xinzweb @dhanashreek89 please advice when you have time.

Jesse and Haisheng
